From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xucy <xucy0609@gmail.com>
Date: Mon, 01 Jan 2026 00:00:00 +0000
Subject: [PATCH] Misc (consolidated)

Consolidated from 1 original patches:
  - 006-Add-BlockPos-transform-to-Position-support.patch


diff --git a/src/main/java/io/papermc/paper/util/MCUtil.java b/src/main/java/io/papermc/paper/util/MCUtil.java
--- a/src/main/java/io/papermc/paper/util/MCUtil.java
+++ b/src/main/java/io/papermc/paper/util/MCUtil.java
@@ -183,6 +183,12 @@
         return CraftLocation.toBukkit(pos, world);
     }
 
+    // Luminol start - add BlockPos support
+    public static Location toLocation(Level world, BlockPos pos) {
+        return new Location(world.getWorld(), pos.getX(), pos.getY(), pos.getZ());
+    }
+    // Luminol end - add BlockPos support
+
     /**
      * Converts a NMS World/Vector to Bukkit Location
      *
-- 

